app/models/miq_group.rb

```diff
@@ -4,7 +4,8 @@ class MiqGroup < ApplicationRecord
   TENANT_GROUP = "tenant"
 
   belongs_to :tenant
-  belongs_to :miq_user_role
+  has_one    :entitlement
+  has_one    :miq_user_role, :through => :entitlement
```

Comment:
yeah, we could probably deprecate and remove things that go through this interface.